src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -225,7 +225,10 @@ private static void EnsureInitialized()
                 // apply the value to the binding
                 if (haveValue && value != null)
                 {
-                    value = ConvertBindingValue(value);
+                    if (!binding.Metadata.Type.Equals("http", StringComparison.OrdinalIgnoreCase))
```

Comment:
Agree with all your points, only added this because I want to keep this scoped down.  It's not specific behavior to http binding, but it's the only one I know I wouldn't break.

The byte[] is converted into a base64 string, and from that point we don't know if it was a user returned base64 string or a byte[] converted string.

I think that it would be useful to pass the objects directly from the function to the BindAsync call to maintain this type info.  If we don't want to make that step, I think I should be able to inspect content type and convert any base64 strings back to byte[] (though that is a lot of wasteful casting).